packages/join-flow/src/pages/payment-details.page.tsx

```diff
@@ -242,3 +252,90 @@ const CreditCardPaymentPage: StagerComponent<FormSchema> = ({
     </form>
   );
 };
+
+const StripePaymentPage: StagerComponent<FormSchema> = ({
+  onCompleted,
+  data
+}) => {
+  const stripePromise = loadStripe(getEnv("STRIPE_PUBLISHABLE_KEY") as string);
+  const plan = (getEnv("MEMBERSHIP_PLANS") as any[]).find(
+    (plan) => plan.value === data.membership
+  );
+  const amount = plan.amount ? plan.amount * 100 : 100;
+  const currency = plan.currency.toLowerCase() || "gbp";
+  const paymentMethodTypes = ['card']
+  if (currency === 'gbp') {
+    paymentMethodTypes.push('bacs_debit')
+  }
+  return (
+    <Elements
+      stripe={stripePromise}
+      options={{
+        paymentMethodCreation: "manual",
+        mode: "subscription",
+        amount,
+        currency,
+        paymentMethodTypes
+      }}
+    >
+      <StripeForm onCompleted={onCompleted} />
+    </Elements>
+  );
+};
+
+const StripeForm = ({
+  onCompleted
+}: {
+  onCompleted: (data: FormSchema) => void;
+}) => {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const handleError = (error: { message?: string }) => {
+    setLoading(false);
+    setErrorMessage(error.message);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    setLoading(true);
+
+    const { error: submitError } = await elements.submit();
+
+    if (submitError) {
+      handleError(submitError);
+      return;
+    }
+
+    const { error, confirmationToken } = await stripe.createConfirmationToken({
+      elements
+    });
+
+    // This point is only reached if there's an immediate error when
+    // creating the ConfirmationToken. Show the error to your customer (for example, payment details incomplete)
```

Comment:
This reads like a copy and paste from Stripe's documentation (which it probably is!). Worth maybe re-phrasing it to fit more cleanly.